layouts/_default/_markup/render-link.html

```diff
@@ -0,0 +1,37 @@
+{{- $link := .Destination }}
+{{- $isRemote := strings.HasPrefix $link "http" }}
+{{- $onlyFragment := strings.HasPrefix $link "#" }}
+{{- $pageLang := .Page.Lang -}}
+
+{{- $filePath := printf "content/%s/%s.md" $pageLang (trim $link "/" ) -}}
+{{- $filePathHasLangPrefix := printf "content/%s.md" (trim $link "/") -}}
+{{- $filePathNoSuffix := printf "content/%s/%s" $pageLang (trim $link "/") -}}
+{{- $filePathHasLangPrefixNoSuffix := printf "content/%s" (trim $link "/") -}}
+
+
+<!-- try link in the lang associated with the current page -->
+
+{{- if $onlyFragment -}}
+<a href="{{ $link | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}>{{ .Text | safeHTML }}</a>
+{{- else -}}
+{{ if $isRemote }}
+<a href="{{ $link | safeURL }}"{{ with .Title}} title="{{ . }}" target="_blank"{{ end }}>{{ .Text | safeHTML }}</a>
+{{- else -}}
+{{- if ne "en" $pageLang -}}
+{{- $checkLink := $link | relLangURL -}}
+{{- if (fileExists $filePath)  -}}
+<a href="{{ $link | relLangURL | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}>USE TRANSLATED:file-exists {{- $filePath -}}{{ .Text | safeHTML }}</a>
+{{- else if (fileExists $filePathHasLangPrefix) -}}
+<a href="{{ $link | relLangURL | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}>USE TRANSLATED:file-exists {{- $filePathHasLangPrefix -}}{{ .Text | safeHTML }}</a>
+{{- else if (fileExists $filePathNoSuffix) -}}
+<a href="{{ $link | relLangURL | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}>USE TRANSLATED:file-exists {{- $filePathNoSuffix -}}{{ .Text | safeHTML }}</a>
+{{- else if (fileExists $filePathHasLangPrefixNoSuffix) -}}
+<a href="{{ $link | relLangURL | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}>USE TRANSLATED:file-exists {{- $filePathHasLangPrefixNoSuffix -}}{{ .Text | safeHTML }}</a>
+{{- else -}}
+<a href="{{ $link | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}>DEFAULT to EN, file does not exist for {{- $link -}}{{ .Text | safeHTML }}</a>
+{{- end -}}
+{{- else -}}
+<a href="{{ $link | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}>EN: {{- $filePath -}} {{ .Text | safeHTML }}</a>
+{{- end -}}
+{{- end -}}
+{{- end -}}
```

Comment:
Tried to digest the logic from line 20 to 37:

```
{{ if $onlyFragment }}
    {{/* skip labels in current page */}}
    <a href="{{ $link | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}>{{ .Text | safeHTML }}</a>
{{ else if $isRemote }}
    {{/* open remote page in new window/tab */}}
    <a href="{{ $link | safeURL }}"{{ with .Title}} title="{{ . }}" target="_blank"{{ end }}>{{ .Text | safeHTML }}</a>
{{ else if eq "en" $pageLang }}
    {{/* skip other English pages */}}
    <a href="{{ $link | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}>{{ .Text | safeHTML }}</a>
{{ else if or (fileExists $filePath) (fileExists $filePathHasLangPrefix) (fileExists $filePathNoSuffix) (fileExists $filePathHasLangPrefixNoSuffix) }}
    {{/* There is a translated version */}}
    {{/* THIS CHECK IS NOT SUFFICIENT FOR REDIRECTED LINKS */}}
    {{/* NOT SURE IF THIS WORKS FOR LINKS TO API REFERENCE URL WHICH CONTAINS GO TEMPLATES ALREADY */}}
    <a href="{{ $link | relLangURL | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}>{{ .Text | safeHTML }}</a>
{{ else }}
    {{/* fall back to English version */}}
    {{/* IS FALLING BACK TO ENGLISH ALWAYS THE BEST OPTION? */}}
    <a href="{{ $link | safeURL }}"{{ with .Title}} title="{{ . }}"{{ end }}>{{ .Text | safeHTML }}</a>
{{ end }}
```

Reply:
Yes, testing/debugging here.